pkg/api/handlers/compat/containers_stats.go

```diff
@@ -56,8 +61,22 @@ func StatsContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	coder := json.NewEncoder(w)
+	// Write header and content type.
+	w.WriteHeader(http.StatusOK)
+	w.Header().Add("Content-Type", "application/json")
+	if flusher, ok := w.(http.Flusher); ok {
+		flusher.Flush()
+	}
+
+	// Setup JSON encoder for streaming.
+	//coder := json.NewEncoder(w)
+	coder.SetEscapeHTML(true)
 	var preRead time.Time
 	var preCPUStats CPUStats
+	if query.oneShot { // if one shot response skip initial read and return one round results
+		goto streamLabel
+	}
```

Comment:
```suggestion
```
Is this test needed? From your check above, `query.Stream` must be false if `query.OneShot` is true.

Reply:
good point, I think I accidentally left this one in from when I had implemented it slightly differently.